altair-viz/dash-vega-components
# Add support for capturing signals and triggering callbacks

This PR adds the possibility to trigger callbacks based on changes in Vega signals, e.g. when a user selects datapoints on a chart, etc. @jonmmease I built this taking quite some inspiration from your JupyterChart implementation in Altair. Any input is greatly appreciated! Also on the Javascript code style. I'd like to merge this latest by Wednesday to cut a new release for a Plotly community meetup on Thursday. No worries if you don't get a chance to look at it until then!

To test it:
* Check out this branch
* `pip install -e .`
* `python example_app.py` to start the example app showcasing the new functionality

Signals are available at the `signalData` property and the user can specify which signals should be listened to using the `signalsToObserve` property. As I use signals directly, this implementation works for Altair, Vega-Lite, and Vega specifications. However, the downside is that it is not aligned with the interfaces provided in https://altair-viz.github.io/user_guide/jupyter_chart.html such as `PointSelection`, `IndexSelection`, ... In my understanding, these would require more logic on the Python side before an Altair chart is converted to a VL spec. 

I'm currently not sure how to do that in Dash and so I'd prefer to start with this general `Vega` component and maybe add a new `Altair` component in a second step which would then only work for Altair charts but could offer the same interface as `altair.JupyterChart`.


https://github.com/altair-viz/dash-vega-components/assets/23366411/4c2545dc-f72a-45fa-9695-a8da287f391b
